Fabric.Authorization.API/scripts/AzureAD-Migration.ps1

```diff
@@ -0,0 +1,252 @@
+#Requires -RunAsAdministrator
+#Requires -Version 5.1
+#Requires -Modules PowerShellGet, PackageManagement
+
+[CmdletBinding()] param(
+    [PSCredential] $credential,
+    [ValidateScript({
+        if (!(Test-Path $_)) {
+            throw "Path $_ does not exist. Please enter valid path to the install.config."
+        }
+        if (!(Test-Path $_ -PathType Leaf)) {
+            throw "Path $_ is not a file. Please enter a valid path to the install.config."
+        }
+        return $true
+    })] 
+    [string] $installConfigPath = "$PSScriptRoot\install.config",
+    [switch] $quiet
+)
+
+Import-Module -Name "$PSScriptRoot\AzureAD-Utilities.psm1" -Force
+Import-Module -Name "$PSScriptRoot\Install-Authorization-Utilities.psm1" -Force
+Import-Module -Name "$PSScriptRoot\Install-Identity-Utilities.psm1" -Force
+Import-Module ActiveDirectory
+
+# Import Fabric Install Utilities
+$fabricInstallUtilities = "$PSScriptRoot\Fabric-Install-Utilities.psm1"
+if (!(Test-Path $fabricInstallUtilities -PathType Leaf)) {
+    Write-DosMessage -Level "Warning" -Message "Could not find fabric install utilities. Manually downloading and installing"
+    Invoke-WebRequest -Uri https://raw.githubusercontent.com/HealthCatalyst/InstallScripts/master/common/Fabric-Install-Utilities.psm1 -Headers @{"Cache-Control" = "no-cache"} -OutFile $fabricInstallUtilities
+}
+Import-Module -Name $fabricInstallUtilities -Force
+
+$credentials = @{}
+
+function Get-NonMigratedActiveDirectoryGroups {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $connectionString
+    )
+
+    $connection = New-Object System.Data.SqlClient.SqlConnection($connectionString)	
+    $sql = "SELECT GroupId, Name, IdentityProvider, ExternalIdentifier, TenantId
+              FROM Groups
+              WHERE IsDeleted = 0
+              AND IdentityProvider = 'Windows'
+              AND Source = 'Directory'"
+    $command = New-Object System.Data.SqlClient.SqlCommand($sql, $connection)
+    
+    Write-DosMessage -Level "Information" -Message "Retrieving groups for migration..."
+
+    $groups = @();
+    try {	
```

Comment:
Not necessary for this PR, but does it make sense to make this a function, and perhaps push it into DosInstallUtilities?